docs/guides/mongodb/clustering/sharding.md

```diff
@@ -0,0 +1,1132 @@
+---
+title: MongoDB Sharding Guide
+menu:
+  docs_0.11.0:
+    identifier: mg-clustering-sharding
+    name: Sharding Guide
+    parent: mg-clustering-mongodb
+    weight: 25
+menu_name: docs_0.11.0
+section_menu_id: guides
+---
+
+> New to KubeDB? Please start [here](/docs/concepts/README.md).
+
+# KubeDB - MongoDB Sharding
+
+This tutorial will show you how to use KubeDB to run a MongoDB Sharding.
+
+## Before You Begin
+
+Before proceeding:
+
+- Read [mongodb sharding concept](/docs/guides/mongodb/clustering/sharding_concept.md) to learn about MongoDB Sharding clustering.
+
+- You need to have a Kubernetes cluster, and the kubectl command-line tool must be configured to communicate with your cluster. If you do not already have a cluster, you can create one by using [Minikube](https://github.com/kubernetes/minikube).
+
+- Now, install KubeDB cli on your workstation and KubeDB operator in your cluster following the steps [here](/docs/setup/install.md).
+
+- To keep things isolated, this tutorial uses a separate namespace called `demo` throughout this tutorial. Run the following command to prepare your cluster for this tutorial:
+
+  ```console
+  $ kubectl create ns demo
+  namespace/demo created
+  ```
+
+> Note: The yaml files used in this tutorial are stored in [docs/examples/mongodb](https://github.com/kubedb/cli/tree/master/docs/examples/mongodb) folder in GitHub repository [kubedb/cli](https://github.com/kubedb/cli).
+
+## Deploy MongoDB Sharding
+
+To deploy a MongoDB Sharding, user have to specify `spec.replicaSet` option in `Mongodb` CRD.
+
+The following is an example of a `Mongodb` object which creates MongoDB Sharding of three members.
+
+```yaml
+apiVersion: kubedb.com/v1alpha1
+kind: MongoDB
+metadata:
+  name: mongo-sh
+  namespace: demo
+spec:
+  version: 3.6-v3
+  shardTopology:
+    configServer:
+      replicas: 3
+      storage:
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+    mongos:
+      replicas: 2
+      strategy:
+        type: RollingUpdate
+    shard:
+      replicas: 3
+      shards: 3
+      storage:
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+```
+
+```console
+$ kubectl create -f https://raw.githubusercontent.com/kubedb/cli/0.11.0//docs/examples/mongodb/clustering/mongo-sharding.yaml
+mongodb.kubedb.com/mongo-sh created
+```
+
+Here,
```

Comment:
Should we use the word `pod` instead of `node` here to clarify that this is k8s pod (or db node, not k8s node)